src/NuGet.Core/NuGet.Versioning/VersionFormatter.cs

```diff
@@ -27,48 +27,24 @@ public string Format(string format, object arg, IFormatProvider formatProvider)
                 throw new ArgumentNullException(nameof(arg));
             }
 
-            string formatted = null;
-            var argType = arg.GetType();
-
-            if (argType == typeof(IFormattable))
+            if (string.IsNullOrEmpty(format) || arg is not SemanticVersion version)
             {
-                formatted = ((IFormattable)arg).ToString(format, formatProvider);
+                return null;
             }
-            else if (!String.IsNullOrEmpty(format))
+
+            var builder = StringBuilderPool.Shared.Rent(Math.Max(format.Length, 100));
+
```

Comment:
Can we use `explicit` type here? Otherwise first time it's not obvious.
I saw other comments about 100 is not important. But for easy to reason why don't we use 256 here?

Reply:
I'll follow up on this.

Reply:
https://github.com/NuGet/NuGet.Client/pull/4146